recipes/orc/all/ConanThirdpartyToolchain.cmake

```
@@ -0,0 +1,49 @@
+# ----------------------------------------------------------------------
+# Snappy
+
+find_package (Snappy REQUIRED)
+
+add_library (orc_snappy INTERFACE)
+add_library (orc::snappy ALIAS orc_snappy)
+target_link_libraries(orc_snappy INTERFACE ${Snappy_LIBRARIES})
+target_include_directories (orc_snappy SYSTEM INTERFACE ${Snappy_INCLUDE_DIR})
```

```suggestion
target_link_libraries(orc_snappy INTERFACE Snappy::snappy)
```

Is not possible to use CMake targets instead? Passing only Snappy_LIBRARIES and Include dir will not cover all aspects, like compiler definitions (in case used). Same for others in this cmake file.

---

No problem, CMake gives a smart way to do that:

```
target_link_libraries(orc_snappy INTERFACE  
    $<$<BOOL:${BUILD_SHARED_LIBS}>:zstd::libzstd_shared>
    $<$<NOT:$<BOOL:${BUILD_SHARED_LIBS}>>:zstd::libzstd_static>
)
```

---

Thanks for providing the generator expression! Unfortunately, it seems that `${BUILD_SHARED_LIBS}` is not enough since 3rd-party dependencies may be specified to different static/shared linkings. For example, following command does not compile:
```
conan create . --version=1.8.6 -o orc/1.8.6:shared=True -o lz4/1.9.4:shared=False
```

---

@wgtmac, good catch! In that case, you could link to the target that's available. As you are using Conan, it will load CMakeDeps, so it should be safe, because only one or another target will be available, no mess, no mixes.

```cmake
target_link_libraries(orc_snappy INTERFACE
    $<TARGET_NAME_IF_EXISTS:zstd::libzstd_static>
    $<TARGET_NAME_IF_EXISTS:zstd::libzstd_shared>
)
```

---

Thanks for the detail guide! Fixed and please take a look again. @uilianries